### ISSUE TITLE
Foreach doesn't work with groups.

### ISSUE THREAD

**Issue by p2rkw** [gittup/tup#223]
Hello,
It seems that groups aren't expanded in rule input, so rules like:
```
: foreach %<group> |> mv %f %o |> %B
```
 doesn't work.

(I'm trying to workaround issue #220 )

Greetings.

**Comment by kavika13** [comment]
You don't need `foreach`. You also have to expand the group using `%<group>` instead of `%f` inside the command.

Try:

     : foreach <group> |> mv %<group> %o |> %B

**Comment by shaneleonard** [comment]
As I understand, this doesn't actually solve the OP's problem though, does it? It appears they want to do:

```
mv foo.c foo
mv bar.c bar
mv baz.c baz
```

The rule suggested by @kavika13 wouldn't even compile, because tup expands the `%<group>` in place, trying to create something like `mv foo.c bar.c baz.c ...`. As a result, `%B` won't work at all because the input is a list of files, therefore `%o` won't expand either and tup will complain.

**Comment by kavika13** [comment]
Yep, my bad. For the `mv` command, that type of group expansion would not work.
